design/src/main/kotlin/uk/govuk/app/design/ui/component/Header.kt

```diff
@@ -40,27 +42,94 @@ fun TabHeader(
     }
 }
 
+data class ActionButton(
+    val text: String? = null,
+    val onClick: () -> Unit
+)
+
 @Composable
 fun ChildPageHeader(
-    text: String,
-    onBack: () -> Unit,
+    text: String? = null,
+    backButton: ActionButton? = null,
+    actionButton: ActionButton? = null,
     modifier: Modifier = Modifier,
 ) {
     Column(modifier) {
-        Box(
-            modifier = Modifier.height(64.dp),
-            contentAlignment = Alignment.CenterStart
-        ) {
-            TextButton(
-                onClick = onBack,
+        if (backButton != null || actionButton != null) {
+            Row(
+                modifier = Modifier
+                    .height(64.dp)
+                    .fillMaxWidth()
+                    .padding(GovUkTheme.spacing.small),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically,
             ) {
-                Icon(
-                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
-                    contentDescription = stringResource(R.string.content_desc_back),
-                    tint = GovUkTheme.colourScheme.textAndIcons.link
+                ChildPageHeaderBackButton(backButton, modifier = modifier.weight(1f))
+                ChildPageHeaderActionButton(
+                    actionButton,
+                    modifier = modifier.weight(1f)
                 )
             }
         }
+
+        if (text != null) {
+            ChildPageHeaderText(text = text, modifier = modifier)
+        }
+    }
+}
+
+@Composable
+private fun ChildPageHeaderBackButton(
+    actionButton: ActionButton? = null,
+    modifier: Modifier = Modifier
+) {
+    if (actionButton == null) return
```

Comment:
I'm not a fan of early returns from functions, what do you guys think?

Reply:
Personally, I like them - it's very Ruby idiomatic to do stuff like this. It also falls nicely into the fail-fast thinking - which I also personally like.

Reply:
It feels a bit weird to add the composable and then emit nothing from it. Rather just not adding it in the first place. It's a subjective thing though so you can leave it in.

Reply:
For me it depends on the function (early return). In this case I would agree with @davidc-gds that it does not feel right to add a composable even if it's redundant?

Reply:
Refactored - see https://github.com/alphagov/govuk-mobile-android-app/pull/142#discussion_r1876308662